scripts/codeowner_analyzer.py

```
@@ -18,6 +18,31 @@
+    Returns a flat dict mapping path prefixes to lists of owners.
+    """
+    with open(filename, "r") as f:
+        raw = json.load(f)
+
+    if not isinstance(raw, list):
+        raise ValueError("Overrides file must contain a JSON array")
+
+    overrides: Dict[str, List[str]] = {}
+    for group in raw:
+        if not isinstance(group, dict) or not isinstance(group.get("owners"), dict):
+            raise ValueError("Each group must be an object with an 'owners' dict")
```

Ruff (0.15.2)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)